crates/zk-por-core/src/recursive_prover/prover.rs

```diff
@@ -12,126 +11,37 @@ use plonky2::{
-
-        recursive_targets.set_targets(
-            pw,
-            self.sub_proofs.to_vec(),
-            &self.merkle_sum_circuit.verifier_only,
-        )
-    }
-
-    /// Gets the proof with pis of this batch of recursive proofs.
-    pub fn get_proof(&self) -> ProofWithPublicInputs<F, C, D>
-    where
-        <C as GenericConfig<2>>::Hasher: AlgebraicHasher<F>,
-    {
-        let mut builder = CircuitBuilder::<F, D>::new(STANDARD_CONFIG);
-        let mut pw = PartialWitness::<F>::new();
-
-        self.build_and_set_recursive_targets(&mut builder, &mut pw);
-
-        builder.print_gate_counts(0);
-
-        let mut timing = TimingTree::new("prove", Level::Info);
-        let data = builder.build::<C>();
-
-        let CircuitData { prover_only, common, verifier_only: _ } = &data;
-
-        log::debug!("before prove");
-        let start = std::time::Instant::now();
-
-        let proof_res = prove(&prover_only, &common, pw.clone(), &mut timing);
-
-        log::debug!("time for {:?} proofs, {:?}", N, start.elapsed().as_millis());
-
-        match proof_res {
-            Ok(proof) => {
-                println!("Finished Proving");
-
-                let proof_verification_res = data.verify(proof.clone());
-                match proof_verification_res {
-                    Ok(_) => proof,
-                    Err(e) => {
-                        error!("Proof verification failed: {:?}", e);
-                        panic!("Proof verification failed!");
-                    }
-                }
-            }
-            Err(e) => {
-                error!("Proof generation failed: {:?}", e);
-                panic!("Proof generation failed!");
-            }
-        }
-    }
-
     /// Get proof with a pre-compiled merkle sum circuit and recursive targets. In this method we do not need to build the circuit as we use a pre-built circuit.
     pub fn get_proof_with_circuit_data(
         &self,
-        recursive_targets: &RecursiveTargets<N>,
+        recursive_targets: RecursiveTargets<N>,
         cd: &CircuitData<F, C, D>,
     ) -> ProofWithPublicInputs<F, C, D>
     where
         <C as GenericConfig<2>>::Hasher: AlgebraicHasher<F>,
     {
         let mut pw = PartialWitness::<F>::new();
-        let CircuitData { prover_only, common, verifier_only } = &cd;
+        let CircuitData { prover_only, common, .. } = &cd;
 
-        recursive_targets.set_targets(&mut pw, self.sub_proofs.to_vec(), verifier_only);
+        recursive_targets.set_targets(&mut pw, self.sub_proofs.to_vec(), &self.sub_circuit_vd);
 
         let mut timing = TimingTree::new("prove", Level::Info);
 
         log::debug!("before prove");
         let start = std::time::Instant::now();
 
-        let proof_res = prove(&prover_only, &common, pw.clone(), &mut timing);
+        let proof_res = prove(&prover_only, &common, pw, &mut timing);
 
         log::debug!("time for {:?} proofs, {:?}", N, start.elapsed().as_millis());
```

Comment:
use debug from tracing, not log. pls help replace

Reply:
ok.